packages/web-components/@sfx/product/test/unit/product.test.ts

```diff
@@ -0,0 +1,199 @@
+    });
+
+    it('should add an eventListener to the window', () => {
+      const addEventListener = spy(window, 'addEventListener');
+
+      component.connectedCallback();
+      expect(addEventListener).to.have.been.calledWith(
+        'sfx::change_product_variant',
+        component.updateVariant
+      );
+    });
+  });
+
+  describe('disconnectedCallback', () => {
+    it('should call its super disconnectedCallback', () => {
+      const disconnectedCallback = stub(Base.prototype, 'disconnectedCallback');
+
+      component.disconnectedCallback();
+      expect(disconnectedCallback).to.have.been.called;
+    });
+
+    it('should remove eventListener from the window', () => {
+      const removeEventListener = spy(window, 'removeEventListener');
+
+      component.disconnectedCallback();
+      expect(removeEventListener).to.have.been.calledWith(
+        'sfx::change_product_variant',
+        component.updateVariant
+      );
+    });
+  });
+
+  describe('updateVariant', () => {
+    it('should update the product property with new info', () => {
+      const event = {
+        detail: { name: 'Foo', price: 12.34 }
+      };
+      component.updateVariant(event);
+      expect(component.product.name).to.equal('Foo');
+      expect(component.product.price).to.equal(12.34);
+    });
+  });
+
+  describe('additionalInfo', () => {
+    it('should return an empty array if there are no additional properties', () => {
+      const result = component.additionalInfo();
+      expect(result).to.eql([]);
+    });
+
+    it('should return an array of TemplateResults there are additional properties', () => {
+      component.product = {
+        name: '',
+        salePrice: 0,
+        promo: '30% off'
+      };
+
+      const result = component.additionalInfo();
+
+      expect(result[1]).to.be.an.instanceOf(TemplateResult);
+    });
+  });
+
+  describe('urlWrap', () => {
+    it('should return the children wrapped in an "a" tag', () => {
+      const result = component.urlWrap('string', html`<span>foo</span>`);
+      expect(result).to.be.an.instanceof(TemplateResult);
+    });
+
+    it('should return the children', () => {
+      const result = component.urlWrap(undefined, html`<span>foo</span>`);
+      expect(result).to.be.an.instanceof(TemplateResult);
+    });
+  });
+
+  describe('render', () => {
+    it('should return an instance of TemplateResult', () => {
+      const result = component.render();
+      expect(result).to.be.an.instanceof(TemplateResult);
+    });
+
+    it('should call the additionalInfo function', () => {
+      const additionalInfo = stub(component, 'additionalInfo')
+      component.render();
+      expect(additionalInfo).to.be.called;
+    });
+
+    it('should call the urlWrap function', () => {
+      const urlWrap = stub(component, 'urlWrap')
+      component.render();
+      expect(urlWrap).to.be.called;
+    });
+
+  });
+})
+
+describe('Variant Component', () => {
```

Comment:
I feel like these tests should be in a different file.

Reply:
re: https://github.com/groupby/sfx-view/pull/8/files#r307448566